src/PasswordStrength/assets/src/controller.ts

```diff
@@ -0,0 +1,127 @@
+import { Controller } from '@hotwired/stimulus';
+
+enum PasswordStrength {
+    STRENGTH_VERY_WEAK = 0,
+    STRENGTH_WEAK = 1,
+    STRENGTH_MEDIUM = 2,
+    STRENGTH_STRONG = 3,
+    STRENGTH_VERY_STRONG = 4,
+}
+
+export default class extends Controller {
+    static values = {
+        veryWeakMessage: { type: String, default: 'The password strength is too weak.' },
+        weakMessage: { type: String, default: 'The password strength is weak.' },
+        mediumMessage: { type: String, default: 'The password strength is medium.' },
+        strongMessage: { type: String, default: 'The password strength is strong.' },
+        veryStrongMessage: { type: String, default: 'The password strength is very strong.' },
+    };
+    static targets: string[] = ['score', 'message', 'meter'];
+
+    declare readonly veryWeakMessageValue: string;
+    declare readonly weakMessageValue: string;
+    declare readonly mediumMessageValue: string;
+    declare readonly strongMessageValue: string;
+    declare readonly veryStrongMessageValue: string;
+    declare readonly scoreTargets: HTMLElement[];
+    declare readonly messageTargets: HTMLElement[];
+    declare readonly meterTargets: HTMLElement[];
+
+    connect() {
+        this.dispatchEvent('connect', {});
+    }
+
+    private dispatchEvent(name: string, payload: any) {
+        this.dispatch(name, { detail: payload, prefix: 'password-strength' });
+    }
+
+    public estimatePasswordStrength(event: InputEvent): void {
+        const password = event.target?.value || '';
+        const length = password.length;
+        const result = {
+            score: PasswordStrength.STRENGTH_VERY_WEAK,
+            message: this.veryWeakMessageValue,
+        };
+
+        if (length !== 0) {
+            const charCount = this.countChars(password);
+            let control = 0;
+            let digit = 0;
+            let upper = 0;
+            let lower = 0;
+            let symbol = 0;
+            let other = 0;
+
+            for (const charCodeStr in charCount) {
+                const charCode = parseInt(charCodeStr);
+                const count = charCount[charCode];
+
+                switch (true) {
+                    case charCode < 32 || charCode === 127:
+                        control += count * 33;
+                        break;
+                    case charCode >= 48 && charCode <= 57:
+                        digit += count * 10;
+                        break;
+                    case charCode >= 65 && charCode <= 90:
+                        upper += count * 26;
+                        break;
+                    case charCode >= 97 && charCode <= 122:
+                        lower += count * 26;
+                        break;
+                    case charCode >= 128:
+                        other += count * 128;
+                        break;
+                    default:
+                        symbol += count * 33;
+                }
```

Comment:
```suggestion

                switch (true) {
                    case charCode < 32 || charCode === 127:
                        control = 33;
                        break;
                    case charCode >= 48 && charCode <= 57:
                        digit = 10;
                        break;
                    case charCode >= 65 && charCode <= 90:
                        upper = 26;
                        break;
                    case charCode >= 97 && charCode <= 122:
                        lower = 26;
                        break;
                    case charCode >= 128:
                        other = 128;
                        break;
                    default:
                        symbol = 33;
                }
```

Number of occurrences of each type of character must not be used to entropy calculation.
See :
- https://github.com/symfony/symfony/blob/5876c48757182da02cbea275a40059b5bfc83854/src/Symfony/Component/Validator/Constraints/PasswordStrengthValidator.php#L65-L77
- https://proton.me/blog/what-is-password-entropy

Reply:
Hello @cvergne,

Indeed the proposal was mot in line with the Symfony constraint.
I will also take the opportunity to rebase this PR.

Regards